modules/oe_paragraphs_timeline/oe_paragraphs_timeline.module

```diff
@@ -23,6 +24,115 @@ function oe_paragraphs_timeline_field_widget_timeline_widget_form_alter(&$elemen
     return;
   }
 
-  $element['label']['#required'] = TRUE;
-  $element['title']['#required'] = TRUE;
+  // We can't use the protected isDefaultValueWidget() method.
+  $is_default_value_widget = (bool) $form_state->get('default_value_widget');
+
+  if (!$is_default_value_widget) {
+    $element['#element_validate'][] = '_oe_paragraphs_validate_required_fields';
+
+    // Get the selector to build the paths for each field.
+    $field_name = $field_definition->getName();
+    $delta = $element['#delta'];
+    $parents = $element['#field_parents'];
+    $parents[] = $field_name;
+    $selector = $root = array_shift($parents);
+    if ($parents) {
+      $selector = $root . '[' . implode('][', $parents) . ']';
+    }
+
+    // Set Label required if Title or Content is filled in.
+    if (!$element['label']['#required']) {
+      $element['label']['#states']['required'][] = [
+        'textarea[name="' . $selector . '[' . $delta . '][body][value]"]' => ['filled' => TRUE],
+      ];
+      $element['label']['#states']['required'][] = [
+        ':input[name="' . $selector . '[' . $delta . '][title]"]' => ['filled' => TRUE],
+      ];
+    }
+    // Set Label required if Title or Content is filled in.
+    if (!$element['title']['#required']) {
+      $element['title']['#states']['required'][] = [
+        'textarea[name="' . $selector . '[' . $delta . '][body][value]"]' => ['filled' => TRUE],
+      ];
+      $element['title']['#states']['required'][] = [
+        ':input[name="' . $selector . '[' . $delta . '][label]"]' => ['filled' => TRUE],
+      ];
+    }
+  }
+}
+
+/**
+ * Validate the required fields of Timeline paragraph.
+ */
+function _oe_paragraphs_validate_required_fields($element, &$form_state, $form) {
+  // Set errors for Label field when any or both of the other 2 fields are
+  // filled in.
+  if ($element['label']['#value'] === '') {
+    if ($element['title']['#value'] !== '' && $element['body']['value']['#value'] !== '') {
```

Comment:
Another note: since we do always the same check, let's cache them in variables:
```
$is_title_empty = $element['title']['#value'] === '';
$is_body_empty = $element['body']['value']['#value'] === '';
$is_label_empty = $element['label']['#value'] === '';
```
(we should use form state but anyway we move now to constraint).

First check will be:
```
if ($is_title_empty && $is_label_empty && $is_body_empty) {
  return;
}
```
Then we can have the next checks for `($is_label_empty && !$is_title_empty)` and `($is_title_empty && !$is_label_empty)`.